### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by SelmaNdapanda**
### **In this branch, I did the following:** :arrow_down: 

- Create a class `Classroom` with the following:  :heavy_check_mark: 
  - `@label` instance variable, should be initialized in the constructor.
  - Setter and getter for `@label` (remember about `attr_accessor`).
- Create the `has-many`/`belongs-to` relationship between `Classroom` and `Student`. The following should be implemented:  :heavy_check_mark: 
  - Create the `has-many` side (a classroom has many students).
  - Create the `belongs-to` side (a student belongs to a classroom).
  - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class `Book` with the following:  :heavy_check_mark: 
  - `@title` and `@author` instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about `attr_accessor`).
- Create a class `Rental` with the following:  :heavy_check_mark: 
  - `@date` instance variable, should be initialized in the constructor.
  - Setter and getter for `@date` (remember about `attr_accessor`).
- Create the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. The following should be implemented:  :heavy_check_mark: 
  - Create the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Create the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Create the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Create the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modify the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 require './nameable'
 
 class Person < Nameable
-  attr_reader :id
+  attr_reader :id, :rentals
   attr_accessor :name, :age
 
   def initialize(age, name = 'Unknown', parent_permission: true)
```
Comment (thecodechaser) [inline person.rb]: - [x] Please make sure in the `Person` class a method for adding rentals should be present and make sure the rentals variable is also writable so move it to `attr_accessor`.
  - You can consider the below code

```ruby
def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_reader :rentals
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (thecodechaser) [inline book.rb]: - [x] Please make sure in `Book` class a method for adding rentals should be present and make sure the rentals variable is also writable so move it to `attr_accessor`.
  - You can consider the below code

```ruby
 def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: classroom.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  attr_reader :students
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    @students.push(student)
+  end
```
Comment (thecodechaser) [inline classroom.rb]: - [x] Please make sure when adding a student to a classroom it also sets the classroom for the student to achieve the `has-many` / `belongs-to` relationships.
  - You can consider the below code for this method.

```ruby
def add_student(student)
    @students.push(student)
    student.classroom = self
  end
```

File: student.rb [hunk 4]
```diff
@@ -1,6 +1,8 @@
 require './person'
 
 class Student < Person
+  attr_accessor :classroom
+
   def initialize(name, age, parent_permission, classroom)
     super(name, age, parent_permission)
     @classroom = classroom
```
Comment (thecodechaser) [inline student.rb]: - [x] Please make sure when setting the classroom for a student it also adds it to the classrooms' students to achieve the `has-many` / `belongs-to` relationships.
  - You can consider adding the below method.

```ruby
def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @SelmaNdapanda👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by SelmaNdapanda** [comment]
This was really helpful. Thank you so much :smile:

**Review by HAMMAS-SALEEM — APPROVED** [review]
# Status: Approved :green_circle: :1st_place_medal: :100:
success

### Hey @SelmaNdapanda 👋

Great work on implementing the requested changes. Your project is complete. :blush:
There is nothing else to say about your project. :+1:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Requirements
✔️ Following Linters

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**